seatunnel-core/seatunnel-flink-starter/seatunnel-flink-13-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkRuntimeEnvironment.java

```diff
@@ -185,157 +82,15 @@ private void createStreamTableEnvironment() {
-                String uri = config.getString(ConfigKeyName.CHECKPOINT_DATA_URI);
-                StateBackend fsStateBackend = new FsStateBackend(uri);
-                if (EnvironmentUtil.hasPathAndWaring(config, ConfigKeyName.STATE_BACKEND)) {
-                    String stateBackend = config.getString(ConfigKeyName.STATE_BACKEND);
-                    if ("rocksdb".equalsIgnoreCase(stateBackend)) {
-                        StateBackend rocksDBStateBackend =
-                                new RocksDBStateBackend(fsStateBackend, TernaryBoolean.TRUE);
-                        environment.setStateBackend(rocksDBStateBackend);
-                    }
-                } else {
-                    environment.setStateBackend(fsStateBackend);
-                }
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.MAX_CONCURRENT_CHECKPOINTS)) {
-                int max = config.getInt(ConfigKeyName.MAX_CONCURRENT_CHECKPOINTS);
-                checkpointConfig.setMaxConcurrentCheckpoints(max);
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(config, ConfigKeyName.CHECKPOINT_CLEANUP_MODE)) {
-                boolean cleanup = config.getBoolean(ConfigKeyName.CHECKPOINT_CLEANUP_MODE);
-                if (cleanup) {
-                    checkpointConfig.enableExternalizedCheckpoints(
-                            CheckpointConfig.ExternalizedCheckpointCleanup.DELETE_ON_CANCELLATION);
-                } else {
-                    checkpointConfig.enableExternalizedCheckpoints(
-                            CheckpointConfig.ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION);
-                }
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.MIN_PAUSE_BETWEEN_CHECKPOINTS)) {
-                long minPause = config.getLong(ConfigKeyName.MIN_PAUSE_BETWEEN_CHECKPOINTS);
-                checkpointConfig.setMinPauseBetweenCheckpoints(minPause);
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.FAIL_ON_CHECKPOINTING_ERRORS)) {
-                int failNum = config.getInt(ConfigKeyName.FAIL_ON_CHECKPOINTING_ERRORS);
-                checkpointConfig.setTolerableCheckpointFailureNumber(failNum);
-            }
-        }
-    }
-
-    public void registerResultTable(Config config, DataStream<Row> dataStream, String name) {
-        StreamTableEnvironment tableEnvironment = this.getStreamTableEnvironment();
-        if (!TableUtil.tableExists(tableEnvironment, name)) {
-            tableEnvironment.createTemporaryView(
-                    name, tableEnvironment.fromChangelogStream(dataStream));
+    protected void setCheckpoint() {
+        super.setCheckpoint();
+        CheckpointConfig checkpointConfig = environment.getCheckpointConfig();
+        if (config.hasPath(EnvCommonOptions.CHECKPOINT_TIMEOUT.key())) {
```

Comment:
why not move set checkpoint timeout to abstract environment？

Reply:
@Hisoka-X  `flink13-starter` use `config.hasPath(EnvCommonOptions.CHECKPOINT_TIMEOUT.key())`
`flink-starter-common` use `EnvironmentUtil.hasPathAndWaring(config, EnvCommonOptions.CHECKPOINT_TIMEOUT.key())`

The method used in the previous code was different. In the `flink-starter-common` model, a log prompt 'log. warn' will be printed（
"the parameter '{}' will be deprecated, please use the 'flink.' prefix with the flink official configuration item to set it",
configKey);`

The developers may have designed it this way, so I didn't make any changes.
If they need to be unified, I can modify it